core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -380,50 +417,39 @@ private[transaction] class TransactionMetadata(val transactionalId: String,
     }
   }
 
-  private def prepareTransitionTo(updatedState: TransactionState,
-                                  updatedProducerId: Long,
-                                  updatedEpoch: Short,
-                                  updatedLastEpoch: Short,
-                                  updatedTxnTimeoutMs: Int,
-                                  updatedTopicPartitions: immutable.Set[TopicPartition],
-                                  updatedTxnStartTimestamp: Long,
-                                  updateTimestamp: Long): TxnTransitMetadata = {
-    prepareTransitionTo(updatedState, updatedProducerId, RecordBatch.NO_PRODUCER_ID, updatedEpoch, updatedLastEpoch, updatedTxnTimeoutMs, updatedTopicPartitions, updatedTxnStartTimestamp, updateTimestamp, TransactionVersion.TV_0)
-  }
-
-  private def prepareTransitionTo(updatedState: TransactionState,
-                                  updatedProducerId: Long,
-                                  nextProducerId: Long,
-                                  updatedEpoch: Short,
-                                  updatedLastEpoch: Short,
-                                  updatedTxnTimeoutMs: Int,
-                                  updatedTopicPartitions: immutable.Set[TopicPartition],
-                                  updatedTxnStartTimestamp: Long,
-                                  updateTimestamp: Long,
-                                  clientTransactionVersion: TransactionVersion): TxnTransitMetadata = {
+  private def prepareTransitionTo(state: TransactionState,
+                                  producerId: Long = this.producerId,
```

Comment:
There was a previous effort to remove default arguments in order to not accidentally set something incorrect. I think the defaults make sense (keeping the same) so I think it makes sense, but wanted to mention this change.

Reply:
The defaults just mean that there is no change, so if the caller doesn't use an argument, then it remains unchanged for the transition.  I think it makes code easier to read than to require every parameter to be specified: the changed stuff is easy to see from just glancing at the function call, no need to mentally parse which arguments are updated and which stay the same.

Reply:
I agree with these points. Was just sharing some historical context on this (and other methods). I think it is valid to use here.